### ISSUE TITLE
CondaFileIOError on Installing * environment... with Miniconda installer on Linux

### ISSUE THREAD

**Issue by rgoubet** [ContinuumIO/anaconda-issues#13110]
### Checklist

- [X] I added a descriptive title.
- [X] I searched through [existing issues](https://github.com/ContinuumIO/anaconda-issues/issues) and couldn't find a solution or duplicate issue.
- [X] I searched on the web (e.g. Google) and didn't find any helpful information.
- [X] I searched the [Anaconda documentation](https://docs.anaconda.com/) and didn't find any helpful information.

### Impacted product

- [ ] A conda package (built by Anaconda)
- [ ] Anaconda Distribution (formerly Anaconda Individual Edition)
- [X] Miniconda
- [ ] Anaconda.org

### What happened?

When executing `sh ./Miniconda3-py310_22.11.1-1-Linux-x86_64.sh` on a fresh installation (having first renamed `~/miniconda3`), I get the following:

 ```
Miniconda3 will now be installed into this location:
/home/user_name/miniconda3

  - Press ENTER to confirm the location
  - Press CTRL-C to abort the installation
  - Or specify a different location below

[/home/user_name/miniconda3] >>> 
PREFIX=/home/user_name/miniconda3
Unpacking payload ...
                                                                                
Installing base environment...


Downloading and Extracting Packages

Preparing transaction: done
Executing transaction: done
./Miniconda3-py310_22.11.1-1-Linux-x86_64.sh: 438: [[: Permission denied

Installing * environment...

./Miniconda3-py310_22.11.1-1-Linux-x86_64.sh: 444: [[: Permission denied

CondaFileIOError: '/home/user_name/miniconda3/pkgs/envs/*/env.txt'. [Errno 2] No such file or directory: '/home/user_name/miniconda3/pkgs/envs/*/env.txt'
```
A `*` environment? No wonder it doesn't work... 

(Note: trying to update my existing environment with the `-u` switch has the same results).

No issue where using the earlier `Miniconda3-py39_4.12.0-Linux-x86_64.sh` package. 

### Expected behavior or outcome

Miniconda should install on Linux

### Conda info

```shell
N/A (fresh install)
```

### Conda config

```shell
N/A (fresh install)
```

### Conda list

```shell
N/A (fresh install)
```

**Comment by dkim** [comment]
Run the installer with `bash` instead of `sh`: `bash Miniconda3-py310_22.11.1-1-Linux-x86_64.sh`. `[[` is an extension from from ksh that bash also supports: https://unix.stackexchange.com/a/306115/.

**Comment by rgoubet** [comment]
Yes indeed! Changing the installation script so that it now requires a specific shell doesn't look like such a great idea, but at least it works, thanks!

**Comment by ashanbrown** [comment]
@rgoubet Should this have been closed?  Do you know if there is a fix coming for the miniconda installation script?  Thanks.

**Comment by rgoubet** [comment]
I think dkim's answer solves. I don't think it was a good idea to make this change, but it's probably not a problem that requires solving.

**Comment by ashanbrown** [comment]
The workaround works for me as well, but I just want to make sure that the request funnels back to the team maintaining to update the install script to make sure that their shebang script is compatible with the scripting language they've chosen.  I'm pretty sure this can be described as a bug.  Assuming this is the right place for that feedback, closing the issue just because there is a workaround, might not get this issue addressed.  I can open a duplicate of this but reopening it seemed the ideal path forward since you've reported it already.

**Comment by hyeokjunejeon** [comment]
When looking into Miniconda3.sh, it still has shebang like this:
```
#!/bin/sh
#
# NAME:  Miniconda3
# VER:   py310_22.11.1-1
# PLAT:  linux-64
# MD5:   81b26c617ffd9011dea96fd57c13e71b
```

Although the suggested workaround works for me, I think it would be better to change the right syntax for that shebang in this script(Miniconda3.sh) by maintainers.

**Comment by SuperiChili** [comment]
Got the same problem 

`Downloading and Extracting Packages

Preparing transaction: done
Executing transaction: done
./Miniconda3-latest-Linux-x86_64.sh: 438: [[: not found

Installing * environment...

./Miniconda3-latest-Linux-x86_64.sh: 444: [[: not found

CondaFileIOError: '/home/razvan/miniconda3/pkgs/envs/*/env.txt'. [Errno 2] No such file or directory: '/home/razvan/miniconda3/pkgs/envs/*/env.txt'
`
and when I try to 

> hat
 I get this error
`[/home/razvan/miniconda3] >>> 
ERROR: File or directory already exists: '/home/razvan/miniconda3'
If you want to update an existing installation, use the -u option.`
How can I fix this? I don't want to update but to finish my miniconda3 installation.

**Comment by uthynauta** [comment]
Don't know why this is not yet solved on overall installation, @dkim workaround solved this for me.

**Comment by rabyj** [comment]
@SuperiChili 
Remove incomplete and bugged install and reinstall with compatible language
```bash
rm -rf /home/razvan/miniconda3
bash /the/installer/script.sh
```

**Comment by pseudoyim** [comment]
@rgoubet, thank you for opening this issue (and everyone else for commenting/responding). This issue should be fixed in the latest release of Miniconda-23.1.0-1, now available here: https://docs.conda.io/en/latest/miniconda.html
